### PR TITLE
fix: second part of price fixes 

### PR DESCRIPTION
**Pull request by lynnshaoyu**
Covers items number 1 and number 4 on this ticket: https://uniswaplabs.atlassian.net/browse/WEB-1487

### CODE REVIEW

File: src/utils/formatNumbers.ts [hunk 1]
```diff
@@ -4,16 +4,32 @@ import { DEFAULT_LOCALE } from 'constants/locales'
...
-  const floatForLargerNumbers = parseFloat(currencyAmount.toFixed(3))
+export const currencyAmountToPreciseFloat = (currencyAmount: CurrencyAmount<Currency> | undefined) => {
+  if (!currencyAmount) return undefined
+  const floatForLargerNumbers = parseFloat(currencyAmount.toExact())
   if (floatForLargerNumbers < 0.1) {
-    return parseFloat(currencyAmount.toSignificant(3))
+    return parseFloat(currencyAmount.toSignificant(6))
   }
-  return floatForLargerNumbers
+  return parseFloat(currencyAmount.toExact())
```
Comment (cmcewen) [inline src/utils/formatNumbers.ts:13]: aren't you doing this above? could you just return `floatForLargerNumbers`?

File: src/utils/formatNumbers.ts [hunk 2]
```diff
@@ -4,16 +4,32 @@ import { DEFAULT_LOCALE } from 'constants/locales'
...
+  digits?: number
+  round?: boolean
 }
 
 // Using a currency library here in case we want to add more in future.
-export const formatDollar = (num: number | undefined | null, isPrice = false, digits = 2, round = true) => {
+export const formatDollar = ({
+  num,
+  isPrice = false,
+  neater = false,
```
Comment (cmcewen) [inline src/utils/formatNumbers.ts:28]: truncateEarly? would prefer a more descriptive term

File: src/components/swap/TradePrice.tsx [hunk 3]
```diff
@@ -41,7 +42,9 @@ export default function TradePrice({ price, showInverted, setShowInverted }: Tra
 
   let formattedPrice: string
   try {
-    formattedPrice = showInverted ? price.toSignificant(4) : price.invert()?.toSignificant(4)
+    formattedPrice = showInverted
+      ? formatTransactionAmount(parseFloat(price.toSignificant(4)))
```
Comment (cmcewen) [inline src/components/swap/TradePrice.tsx:46]: are we losing precision here?

File: src/utils/formatNumbers.ts (hunk 1 above)
Comment (lynnshaoyu) [reply src/utils/formatNumbers.ts:13]: yes omg sorry, rushed botched copy pasta.

File: src/utils/formatNumbers.ts (hunk 2 above)
Comment (lynnshaoyu) [reply src/utils/formatNumbers.ts:28]: used lessPreciseStablecoinValues and added a comment to explain

File: src/components/swap/TradePrice.tsx (hunk 3 above)
Comment (lynnshaoyu) [reply src/components/swap/TradePrice.tsx:46]: good point, aded new fn priceToPreciseFloat

**Review by cmcewen — APPROVED** [review]
woot

**Pull request merged**